### ISSUE TITLE
CustomStreamer sample can print replacement character �

### ISSUE THREAD

**Issue by e87tn95h** [openvinotoolkit/openvino.genai#1381]
The following comparison in [multinomial_causal_lm](https://github.com/openvinotoolkit/openvino.genai/tree/releases/2024/3/samples/python/multinomial_causal_lm/README.md) sample as CustomStreamer sample Python code does not make sense. It never match because two string (`text[-3:]` and `chr(65533)`) lengths do not match.

https://github.com/openvinotoolkit/openvino.genai/blob/d17f7168f278ef98acfdc7ba1ac93e4c759a6402/samples/python/multinomial_causal_lm/multinomial_causal_lm.py#L93-L95

It looks like that `elif text[-1] == chr(65533):` is enough to detect � (U+FFFD: replacement character) for this case in Python.

**Comment by pavel-esir** [comment]
@e87tn95h you are right. Without much consideration we copied logic from c++ here, but it was wrong and you are right.  `�` is 3 bytes long, but len('�') == 1. Thanks for noticing this!

Will you open PR yourself or should we fix it?

**Comment by e87tn95h** [comment]
I don't have PR, also I won't be able to make PR for this project as speedy as you.
If this shared idea is good for you and your project, please feel free taking it (and then close this issue).

Thank you for reading and regards,
